src/controllers/sdex/index.ts

```diff
@@ -29,12 +39,23 @@ router.get(
   }),
 );
 
-router.get('/user_pool_positions', async (req, res) => {
-  res.status(200).json({
-    data: {
-      user_pool_positions: [],
-    },
-  });
-});
+router.get(
+  '/user_pool_positions',
+  asyncRoute(async (req, res) => {
+    const { user, chainId, base, quote, poolIdx } = validate(querySchema, req.query);
+
+    return maybeCacheResponse(
+      res,
+      `sdex/user_pool_positions/${chainId}/${user}`,
```

Comment:
Put all identifiers to cache key (otherwise user will see same same balances for each pools until cache expires)

```suggestion
     `sdex/user_pool_positions/${chainId}/${user}/${base}/${quote}/${poolIdx}`,
```